recipes/argon2/all/test_package/conanfile.py

```diff
@@ -0,0 +1,17 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class Argon2TestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sargon2-test" % os.sep)
```

Comment:
There is a better way to handle dynamic deps:

```suggestion
            bin_path = os.path.join("bin", "argon2-test")
            self.run(bin_path, run_environment=True)
```